### DISCUSSION TITLE
How to install `pip` dependencies for VS plugins?

### DISCUSSION THREAD

**Discussion by arthur-st** [staxrip/staxrip#1111] [Q&A] [answered]
https://github.com/HolyWu/vs-basicvsrpp

I'd like to run this, but it seems that VapourSynth frame server embedded with StaxRip has an embedded Python runtime that's not really usable normally. What is the supported solution for installing arbitrary dependencies via `pip`?

**Comment by jessielw** [comment]
StaxRip is using an embedded python version that doesn't support pip without work around. 

However you should be able to manually add anything you need to use.

**Reply by arthur-st** [reply]
Manually copying and pasting site packages, if that's you're suggesting, is not a sustainable workflow. That said, I appreciate the confirmation that one shouldn't try to work the embedded Python distribution for the purposes of adding new packages.

**Comment by arthur-st** [comment, marked as answer]
I have solved this for myself by installing VapourSynth separately and configuring the standalone VapourSynth to use a standard Python environment I have installed in my system, and setting StaxRip to use these two instead of the embedded.

**Reply by jessielw** [reply]
Unfortunately, the embedded python is the limitation here and not StaxRip. There is work arounds and even ways to make pip work, but the solution you did is the correct way.

**Reply by arthur-st** [reply]
Oh, I recognize that this is not a problem created by StaxRip. I was just wondering if StaxRip has a built-in solution of it that is less involved than what I ended up doing, after I fruitlessly tried the officially supported methods for bootstrapping `pip`, `ensurepip` and `get-pip.py`, for the embedded Python distribution.
